plugins/zoom/icon_zoom/util/api.py

```diff
@@ -152,27 +152,37 @@ def _call_api(
             raise PluginException(preset=PluginException.Preset.UNKNOWN)
 
     def _handle_response(self, response: Response, allow_404: bool):
+        """
+        Helper function to process the response based on the status code returned.
+        :param response: Response object
+        :param allow_404: Boolean value to indicate whether to allow 404 status code to be ignored
+        """
+
+        if response.status_code == 400:
+            raise PluginException(preset=PluginException.Preset.BAD_REQUEST, data=response.json())
+
         if response.status_code == 401:
             resp = json.loads(response.text)
             raise PluginException(
                 cause=resp.get("message"),
                 assistance="Verify that the credentials configured within the Zoom plugin connection are correct.",
             )
+
         if response.status_code == 404:
-            resp = json.loads(response.text)
             if allow_404:
                 return None
             else:
-                raise PluginException(
-                    cause=resp.get("message"),
-                    assistance=f"The object at {response.url} does not exist. Verify the ID and fields "
-                    f"used are valid.",
-                )
+                raise PluginException(preset=PluginException.Preset.NOT_FOUND, data=response.json())
+
+        if response.status_code == 409:
```

Comment:
Currently `409` is only returned for one action, `create_user`, so currently this message is relevant until other actions are ever added that also return a `409` status code.

The data is returned but it is within the `cause` variable - let me know what you think, if we should move it to data instead and add something else into `cause`.

Reply:
I think data is a better place for the response since it's typically how we utilise PluginException. I think we can maybe also reword assistance to not prompt the user to try again. "Please check your input is correct and try again" or something along those lines?